### DISCUSSION TITLE
Save image using Laravel's Storage facade

### DISCUSSION THREAD

**Discussion by iDGE90** [spatie/image#173] [General]
At this point there is no way to opt out of the default image saving pattern which uses php's `copy` function. It would be a nice addition to be able to manually save the processed image, for ex:
```
Image::load($image)
    ->fit(Manipulations::FIT_CROP, 150, 150)
    ->format(Manipulations::FORMAT_JPG)
    ->manualSave(function($imageContent) {
        Storage::disk('public')->put('some/path/file.jpg', $imageContent);
    });
```
I'm guessing additional issue would be image optimization which is executed after image is saved.

**Comment by freekmurze** [comment]
I see this as out of scope for this package. For more full media handling capabilities check out our laravel-medialibrary package.

**Comment by mferrara** [comment]
I'm gonna give this a +1 - I'm honestly pretty dumbfounded that there's no way to 'save' to a string/buffer. 

My use case is that I want to make manipulations to images and immediately push those to cloud storage, seems sub-optimal to save them to disk just read them again and then delete them.

Am I missing something?

**Reply by valentinhttr** [reply]
Hi, do you have find a solution?

**Reply by mferrara** [reply]
@valh1996 - I didn't, I'm still persisting to disk and then pushing to the next operation.

**Comment by indigoram89** [comment]
Is it still really impossible to use a cloud storage??

**Reply by vpuentem** [reply]
+1 same question

**Reply by lbattaglioli2000** [reply]
I've been trying to use this library within the context of Laravel Vapor, and because I can't touch the filesystem for image storage, I haven't been able to find a good way to use this package, unfortunately.

**Reply by mwolff-fn** [reply]
Almost two years later I was asking myself the same thing - and while I haven't decided on using this package yet here's a suggestion in case anyone else stumbles upon this thread: Short of forking this package and adding the required functionality, it may be easier to just write a custom stream handler and then pass the appropriate stream URL as path, e.g. "mygloriouscloudstorage:///path/on/cloud/storage/myimage.png". Immediately works with all PHP file functions, although not sure if the Imagick, GD etc. extensions use these internally.
